pallets/weights/src/pallet_settlement.rs

```diff
@@ -469,4 +470,126 @@ impl pallet_settlement::WeightInfo for WeightInfo {
             .saturating_add(DbWeight::get().writes(6 as Weight))
             .saturating_add(DbWeight::get().writes((16 as Weight).saturating_mul(l as Weight)))
     }
+
+    // Storage: Identity KeyRecords (r:1 w:0)
+    // Storage: Settlement VenueInfo (r:1 w:0)
+    // Storage: Settlement VenueFiltering (r:2 w:0)
+    // Storage: Settlement InstructionCounter (r:1 w:1)
+    // Storage: Timestamp Now (r:1 w:0)
+    // Storage: Scheduler Lookup (r:1 w:1)
+    // Storage: Scheduler Agenda (r:1 w:1)
+    // Storage: Settlement UserAffirmations (r:0 w:2)
+    // Storage: Settlement InstructionLegsV2 (r:0 w:11)
+    // Storage: Settlement InstructionAffirmsPending (r:0 w:1)
+    // Storage: Settlement InstructionMemos (r:0 w:1)
+    // Storage: Settlement InstructionDetails (r:0 w:1)
+    // Storage: Settlement VenueInstructions (r:0 w:1)
+    fn add_instruction_with_memo_v2(f: u32) -> Weight {
+        (101_933_000 as Weight)
+            .saturating_add(DbWeight::get().reads(8 as Weight))
+            .saturating_add(DbWeight::get().writes(19 as Weight))
+            .saturating_add(DbWeight::get().writes((1 as Weight).saturating_mul(f as Weight)))
+    }
+
+    // Storage: Identity KeyRecords (r:1 w:0)
+    // Storage: unknown [0x3a7472616e73616374696f6e5f6c6576656c3a] (r:1 w:1)
+    // Storage: Settlement VenueInfo (r:1 w:0)
+    // Storage: Settlement VenueFiltering (r:2 w:0)
+    // Storage: Settlement InstructionCounter (r:1 w:1)
+    // Storage: Timestamp Now (r:1 w:0)
+    // Storage: Scheduler Lookup (r:1 w:1)
+    // Storage: Scheduler Agenda (r:1 w:1)
+    // Storage: Portfolio PortfolioCustodian (r:1 w:0)
+    // Storage: Settlement InstructionLegsV2 (r:12 w:11)
+    // Storage: Portfolio PortfolioNFT (r:100 w:0)
+    // Storage: Portfolio PortfolioLockedNFT (r:100 w:100)
+    // Storage: Asset Tokens (r:1 w:0)
+    // Storage: Portfolio PortfolioAssetBalances (r:1 w:0)
+    // Storage: Portfolio PortfolioLockedAssets (r:1 w:1)
+    // Storage: Settlement UserAffirmations (r:0 w:2)
+    // Storage: Settlement InstructionAffirmsPending (r:0 w:1)
+    // Storage: Settlement InstructionMemos (r:0 w:1)
+    // Storage: Settlement InstructionDetails (r:0 w:1)
+    // Storage: Settlement VenueInstructions (r:0 w:1)
+    // Storage: Settlement AffirmsReceived (r:0 w:1)
+    // Storage: Settlement InstructionLegStatus (r:0 w:11)
+    fn add_and_affirm_instruction_with_memo_v2(legs_v2: &[LegV2]) -> Weight {
+        let (f, n) = get_transfer_by_asset(legs_v2);
```

Comment:
The benchmarks can't generate this weight function.  But a helper method can be added to `trait pallet_settlement::WeightInfo` that takes the legs and calculates the `f, n` parameters to pass to the generated weight function.

Add to the trait in `pallets/settlement/src/lib.rs`:
```
    fn add_and_affirm_instruction_with_memo_v2_legs(legs_v2: &[LegV2]) -> Weight {
        let (f, n) = get_transfer_by_asset(legs_v2);
        Self::add_and_affirm_instruction_with_memo_v2(f, n)
    }
```

```suggestion
    fn add_and_affirm_instruction_with_memo_v2(f: u32, n: u32) -> Weight {
```